runtime/compiler/optimizer/FearPointAnalysis.cpp

```diff
@@ -35,16 +35,26 @@
 #include "infra/Checklist.hpp"
 
 
-bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp, TR::Node *virtualGuardNode)
+bool TR_FearPointAnalysis::virtualGuardKillFear(TR::Compilation *comp, TR::Node *virtualGuardNode)
    {
+   if (comp->cg()->supportsMergingGuards())
+      return false;
+   
    static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear")) == NULL;
-   if (kill && !comp->getOption(TR_DisableVectorAPIExpansion))
+
+   if (kill && !comp->getOption(TR_DisableVectorAPIExpansion) && comp->getMethodSymbol()->hasVectorAPI())
       {
       TR_VirtualGuard *guardInfo = comp->findVirtualGuardInfo(virtualGuardNode);
-      if (guardInfo != NULL && guardInfo->isInlineGuard())
+      if (guardInfo->isInlineGuard())
          {
-         char* methodSig = guardInfo->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethod()->classNameChars();
-         return (strncmp("jdk/incubator/vector", methodSig, 20) != 0 && strncmp("jdk/internal/vm/vector", methodSig, 22) != 0);
+         uint32_t classNameLength = guardInfo->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethod()->classNameLength();
+         char* className = guardInfo->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethod()->classNameChars();
```

Comment:
Since other changes are still needed, would you mind factoring out `guardInfo->getSymbolReference()->getSymbol()->castToMethodSymbol()->getMethod()` into a variable so that it doesn't have to be repeated?

Reply:
Fixed in https://github.com/eclipse-openj9/openj9/pull/16354/commits/9c03ca552e5d1d569099659854a525442bc9ee0a